### PR TITLE
doc: additional scalaDoc for event stream semantics (#26854)

### PR DESCRIPTION
**Pull request by Roiocam**
References #26854, I hope my understanding is correct.

### CODE REVIEW

File: akka-actor-typed/src/main/scala/akka/actor/typed/eventstream/EventStream.scala [hunk 1]
```diff
@@ -61,6 +65,14 @@ object EventStream {
...
    * Unsubscribe an actor ref from the event stream
    * by sending this command to the [[akka.actor.typed.ActorSystem.eventStream]].
+   * <p>
+   *  <strong>
+   *   Unsubscribing an unsubscribed actor ref will triggers the expensive
+   * check of "whether actor ref still has subscriptions" in the [[akka.event.EventStream]],
+   * even though the operation is not performed on the EventStream itself,
+   * but it affects the timely "watch" of new subscribers by the [[akka.event.EventStream]].
+   *  </strong>
+   * </p>
```
Comment (johanandren) [inline akka-actor-typed/src/main/scala/akka/actor/typed/eventstream/EventStream.scala:75]: The most  interesting part here is the semantics, that the actor unsubscribes to all subscriptions created for it, not entirely sure the performance caveeat is needed.

File: akka-actor-typed/src/main/scala/akka/actor/typed/eventstream/EventStream.scala [hunk 2]
```diff
@@ -31,6 +31,10 @@ object EventStream {
   /**
    * Subscribe a typed actor to listen for types or subtypes of E
    * by sending this command to the [[akka.actor.typed.ActorSystem.eventStream]].
+   * <p>
+   *   Subscribe several times via send multiple Subscribe(a) are allowed because the
+   * EventStream's watch operation on an Actor is idempotent.
+   * </p>
```
Comment (johanandren) [inline akka-actor-typed/src/main/scala/akka/actor/typed/eventstream/EventStream.scala:37]: The most important part here is that the same actor can create multiple subscriptions to different types of published messages.

**Review by johanandren — COMMENTED** [review]
Note that the issue also mentions that we have no reference docs for the typed event stream. Fine to just improve the API-docs here but we should leave the issue open for those once this is done.

**Comment by Roiocam** [comment]
> Note that the issue also mentions that we have no reference docs for the typed event stream. Fine to just improve the API-docs here but we should leave the issue open for those once this is done.

Yep. The original thought of this PR came from #29109, and I have been using typed event streams for a while now. However, I noticed that it still lacks documentation. I will fix those semantics later.

One more thing,I have some interesting use cases for the event stream:

- Collect all exceptions, save it to memory, and display them on the UI for diagnostics.
- Publish events on certain Actor lifecycles, collect them, and display them on the UI. This way, we can obtain information about "who is alive" and "where they are alive".

**Comment by Roiocam** [comment]
Oops, there's one more thing. The typed event stream document should be retained the `EventBus` document, or it can simply be named as`EventStream` and placed under`Utilities`.

**Comment by johanandren** [comment]
We should keep the event bus page but have the same kind of header cross-linking to typed as we have for other features that exist in both classic and typed.

**Review by johanandren — APPROVED** [review]
LGTM, thanks!

**Comment by johanandren** [comment]
(close-reopen to trigger bincomp check on Scala 3.3)

**Pull request merged**